batchflow/models/torch/base.py

```diff
@@ -1070,14 +1080,20 @@ def _train(self, *args, fetches=None, train_mode='', accumulate_grads=True, sync
                 else:
                     self.sync_counter += 1
 
-                if decay:
-                    if step.get('current_iter', 0) >= step['n_iters']:
-                        decay.step()
-                        step['current_iter'] = 0
-                    step['current_iter'] = step.get('current_iter', 0) + 1
+                curr_lr = [group['lr'] for group in optimizer.param_groups]
+                self.iter_info.setdefault('lr', []).append(curr_lr)
+
+                if decays:
+                    for decay, decay_step in zip(decays, decay_steps):
+                        step_condition = (step['iter'] - decay_step['first_iter']) % decay_step['frequency'] == 0
+                        border_condition = (step['iter'] >= decay_step['first_iter'] and
```

Comment:
Mention that `first_iter` and `last_iter` are included in the range of iterations on which the decay is applied